src/models/hooks/goalFieldResponse.js

```diff
@@ -56,20 +56,42 @@ const syncActivityReportGoalFieldResponses = async (sequelize, instance, _option
       }],
     });
 
-    const activityReportIds = activityReportGoals.map((item) => item.activityReportId);
-    const activityReportGoalIds = activityReportGoals.map((item) => item.id);
+    const activityReportIds = [
+      ...new Set(activityReportGoals.map((item) => item.activityReportId)),
+    ];
+    const activityReportGoalIds = [...new Set(activityReportGoals.map((item) => item.id))];
 
     // Perform the update.
-    await sequelize.models.ActivityReportGoalFieldResponse.update(
+    const updatedResponses = await sequelize.models.ActivityReportGoalFieldResponse.update(
       { response: instance.response },
       {
         where: {
           activityReportGoalId: activityReportGoalIds,
           goalTemplateFieldPromptId,
         },
+        returning: true,
       },
     );
 
+    // Get all the ActivityReportGoal ids that were updated.
+    const updatedResponseIds = [
+      ...new Set(updatedResponses[1].map((item) => item.activityReportGoalId)),
+    ];
+
+    // If the activityReportGoal exists but wasn't updated we know its missing a response.
```

Comment:
Is this always a true statement?

Reply:
Ok I will write a test for this case to ensure if its already the value in the update it still comes back in the updated list.

Reply:
I have added a test for this case.